src/groups/group_action.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    AbstractActionOnManifold
+
+An abstract group action on a manifold.
+"""
+abstract type AbstractActionOnManifold end
+
+"""
+    base_group(A::AbstractActionOnManifold)
+
+The group that acts in action `A`.
+"""
+function base_group(A::AbstractActionOnManifold)
+    error("Function base_group is not yet defined for type $(typeof(A)).")
+end
+
+"""
+    action_on(A::AbstractActionOnManifold)
+
+The manifold the action `A` acts upon.
+"""
+function action_on(A::AbstractActionOnManifold)
+    error("Function action_on is not yet defined for type $(typeof(A)).")
+end
+
+"""
+    apply_action(A::ActionOnManifold, x, a)
+
+Apply action `a` of type `A` to the point `x`.
+The result is saved in `y`.
+"""
+function apply_action!(A::AbstractActionOnManifold, y, x, a)
+    error("Function apply_action! is not yet defined for types $(typeof(A)), $(typeof(y)), $(typeof(x)) and $(typeof(a)).")
+end
+
+"""
+    apply_action(A::AbstractActionOnManifold, x, a)
+
+Apply action `a` of type `A` to the point `x`.
+"""
+function apply_action(A::AbstractActionOnManifold, x, a)
+    y = similar_result(A, apply_action, x, a)
+    apply_action!(A, y, x)
+    return y
+end
+
+"""
+    optimal_alignment(A::AbstractActionOnManifold, x1, x2)
+
+Calculate an action element of action `A` that, when acted
+upon `x1`, gets the element closest to `x2`.
```

Comment:
I prefer "... that acts upon `x1` to produce the element closest to `x2`."
Also, by what definition of "closest"? Using the metric on the group manifold or the g-manifold? Or something else?

Reply:
OK, I'll rephrase it. The metric from the g-manifold is used to define "closest". I'll add it to the docstring.